benchmark/blas/blas.cpp

```diff
@@ -0,0 +1,540 @@
+                 exec, dims.n, dims.r, dims.stride_x, dims.stride_y);
+         }},
+        {"norm",
+         [](std::shared_ptr<const gko::Executor> exec, dimensions dims) {
+             return std::make_unique<NormOperation>(exec, dims.n, dims.r,
+                                                    dims.stride_y);
+         }},
+        {"mm",
+         [](std::shared_ptr<const gko::Executor> exec, dimensions dims) {
+             return std::make_unique<ApplyOperation>(
+                 exec, dims.n, dims.k, dims.m, dims.stride_A, dims.stride_B,
+                 dims.stride_C);
+         }},
+        {"gemm",
+         [](std::shared_ptr<const gko::Executor> exec, dimensions dims) {
+             return std::make_unique<AdvancedApplyOperation>(
+                 exec, dims.n, dims.k, dims.m, dims.stride_A, dims.stride_B,
+                 dims.stride_C);
+         }}};
+
+
+void apply_blas(const char *operation_name, std::shared_ptr<gko::Executor> exec,
+                rapidjson::Value &test_case,
+                rapidjson::MemoryPoolAllocator<> &allocator)
+{
+    try {
+        auto &blas_case = test_case["blas"];
+        add_or_set_member(blas_case, operation_name,
+                          rapidjson::Value(rapidjson::kObjectType), allocator);
+
+        auto op = operation_map[operation_name](exec, parse_dims(test_case));
+
+        // warm run
+        for (unsigned int i = 0; i < FLAGS_warmup; i++) {
+            exec->synchronize();
+            op->run();
+            exec->synchronize();
+        }
+
+        // timed run
+        auto timer = get_timer(exec, FLAGS_gpu_timer);
+        for (unsigned int i = 0; i < FLAGS_repetitions; i++) {
+            exec->synchronize();
+            timer->tic();
+            op->run();
+            timer->toc();
+        }
+        auto runtime = timer->compute_average_time();
+        auto flops = static_cast<double>(op->get_flops());
+        auto mem = static_cast<double>(op->get_memory());
+        add_or_set_member(blas_case[operation_name], "time", runtime,
+                          allocator);
+        add_or_set_member(blas_case[operation_name], "flops", flops / runtime,
+                          allocator);
+        add_or_set_member(blas_case[operation_name], "bandwidth", mem / runtime,
+                          allocator);
+
+        // compute and write benchmark data
+        add_or_set_member(blas_case[operation_name], "completed", true,
+                          allocator);
+    } catch (const std::exception &e) {
+        add_or_set_member(test_case["blas"][operation_name], "completed", false,
+                          allocator);
+        std::cerr << "Error when processing test case " << test_case << "\n"
+                  << "what(): " << e.what() << std::endl;
+    }
+}
+
+
+int main(int argc, char *argv[])
+{
+    std::string header =
+        "A benchmark for measuring performance of Ginkgo's BLAS-like "
+        "operations.\n";
```

Comment:
Could you write what arguments should be in json? I think it is fine to combine this with the doc from gflags.